game.js

```diff
@@ -0,0 +1,138 @@
+let helpers = require("./helpers.js");
+let lastLetter = "";
+let cities = [
+  "Архангельск",
+  "Борисполь",
+  "Воронеж",
+  "Гюмри",
+  "Донецк",
+  "Енакиево",
+  "Жуковский",
+  "Запорожье",
+  "Изюм",
+  "Йошкар-Ола",
+  "Киев",
+  "Лондон",
+  "Москва",
+  "Николаев",
+  "Орёл",
+  "Париж",
+  "Ростов",
+  "Селидово",
+  "Трускавец",
+  "Уфа",
+  "Флоренция",
+  "Харьков",
+  "Цюрих",
+  "Челябинск",
+  "Шанхай",
+  "Щёлково",
+  "Ыгдыр",
+  "Электросталь",
+  "Юрмала",
+  "Ялта"
+].map(item => item.toLowerCase());
+
+let sessions = {
+  makeSession(chatID) {
+    this[chatID] = { spentCities: new Set() };
+  },
+  deleteSession(chatID) {
+    delete this[chatID];
+  }
+};
+
+function randomCity(arrCities) {
+  return arrCities[helpers.getRandomNumber(arrCities.length)];
+}
+
+function lastValidLetter(str) {
+  let lastLetter = str[str.length - 1];
+  let invalidLetters = ["ь", "ъ"];
+  let i = 2;
+
+  while (invalidLetters.includes(lastLetter)) {
+    lastLetter = str[str.length - i];
```

Comment:
what if there will be no valid letter at all? "ьььььь" for example

Reply:
If we consider this function in the context of the entire program, then such a case will not arise at all. Since the check above eliminates words beginning with invalid letters. You can of course also add a check for all invalid letters in the word. The question is, will it be redundant in the code?